cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,337 @@
+		web: Data::Raw(b"https://xcm-test.io".to_vec().try_into().unwrap()),
+		matrix: Data::Raw(b"riot-xcm-test".to_vec().try_into().unwrap()),
+		email: Data::Raw(b"xcm-test@gmail.com".to_vec().try_into().unwrap()),
+		pgp_fingerprint: Some(pgp_fingerprint),
+		image: Data::Raw(b"xcm-test.png".to_vec().try_into().unwrap()),
+		twitter: Data::Raw(b"@xcm-test".to_vec().try_into().unwrap()),
+		github: Data::None,
+		discord: Data::None,
+	};
+	// Full Identity with nonsensical additional
+	let mut third_relay = second_relay.clone();
+	third_relay.additional = BoundedVec::try_from(vec![(
+		Data::Raw(b"foO".to_vec().try_into().unwrap()),
+		Data::Raw(b"bAr".to_vec().try_into().unwrap()),
+	)])
+	.unwrap();
+	// Full Identity with meaningful additional
+	let mut fourth_relay = second_relay.clone();
+	fourth_relay.additional = BoundedVec::try_from(vec![
+		(
+			Data::Raw(b"github".to_vec().try_into().unwrap()),
+			Data::Raw(b"niels-username".to_vec().try_into().unwrap()),
+		),
+		(
+			Data::Raw(b"discord".to_vec().try_into().unwrap()),
+			Data::Raw(b"bohr-username".to_vec().try_into().unwrap()),
+		),
+	])
+	.unwrap();
+	let mut fourth_para = second_para.clone();
+	fourth_para.github = Data::Raw(b"niels-usernmae".to_vec().try_into().unwrap());
+	fourth_para.discord = Data::Raw(b"bohr-username".to_vec().try_into().unwrap());
+
+	vec![
+		Identity { relay: first_relay, para: first_para },
+		Identity { relay: second_relay, para: second_para.clone() },
+		Identity { relay: third_relay, para: second_para.clone() }, // same as second_para
+		Identity { relay: fourth_relay, para: fourth_para },
+	]
+}
+
+fn id_deposit_parachain(id: &IdentityInfoParachain) -> Balance {
+	let base_deposit = BasicDepositParachain::get();
+	let byte_deposit =
+		ByteDepositParachain::get() * TryInto::<u128>::try_into(id.encoded_size()).unwrap();
+	base_deposit + byte_deposit
+}
+
+fn id_deposit_relaychain(id: &IdentityInfo<MaxAdditionalFields>) -> Balance {
+	let base_deposit = BasicDeposit::get();
+	let byte_deposit = ByteDeposit::get() * TryInto::<u128>::try_into(id.encoded_size()).unwrap();
+	base_deposit + byte_deposit
+}
+
+fn set_id_relay(id: &Identity) -> u128 {
+	let mut total_deposit = 0_u128;
+
+	// Set identity and Subs on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+
+		// 1. Set identity on Relay Chain
+		assert_ok!(RococoIdentity::set_identity(
+			RococoOrigin::signed(RococoRelaySender::get()),
+			Box::new(id.relay.clone())
+		));
+
+		// 2. Set sub-identity on Relay Chain
+		assert_ok!(RococoIdentity::set_subs(
+			RococoOrigin::signed(RococoRelaySender::get()),
+			vec![(RococoRelayReceiver::get(), Data::Raw(vec![42; 1].try_into().unwrap()))],
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Identity(IdentityEvent::IdentitySet { .. }) => {},
+				RuntimeEvent::Balances(BalancesEvent::Reserved { .. }) => {},
```

Comment:
We should expect two `Reserved` events, one for the identity and one for the sub.